esx_service/utils/auth_api.py

```diff
@@ -365,14 +366,81 @@ def is_vm_duplicate(vm_list):
+    # The valid default_datastore name are:
+    # named datastore existing on the host
+    # hard coded datastore name "__VM_DS"
+    # "__ALL_DS" is not a valid value to set as "default_datastore"
+    if datastore_name == auth_data_const.VM_DS:
+        return None
+    if datastore_name == auth_data_const.ALL_DS:
+        return generate_error_info(ErrorCode.DS_DEFAULT_NAME_INVALID, datastore_name)
+
+    if not vmdk_utils.validate_datastore(datastore_name):
+        error_info = generate_error_info(ErrorCode.DS_NOT_EXIST, datastore_name)
+        return error_info
+
+    return None
+
+def set_default_ds_and_create_privilege(tenant, default_datastore, check_existing):
+    """
+        Set "default_datastore" for given tenant and create a full access privilege
+        to "default_datastore" if entry does not exist
+    """
+    # @Param tenant is a DockerVolumeTenant object
+    logging.debug("set_default_ds_and_create_privilege: tenant_name=%s default_datastore=%s check_existing=%d",
+                  tenant.name, default_datastore, check_existing)
+
+    error_info, auth_mgr = get_auth_mgr_object()
+    if error_info:
+        return error_info
+
+    datastore_url = vmdk_utils.get_datastore_url(default_datastore)
+    # datastore_url will be set to "None" by "vmdk_utils.get_datastore_url" is "default_datastore"
+    # is not a valid datastore
+    if datastore_url is None:
+        error_info = generate_error_info(ErrorCode.DS_DEFAULT_NAME_INVALID, default_datastore)
+        return error_info
+
+    existing_default_ds_url = None
+    if check_existing:
+        error_msg, existing_default_ds_url = tenant.get_default_datastore(auth_mgr.conn)
+        if error_msg:
+            error_info = generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
+            return error_info
+
+        # the "default_datastore" to be set is the same as existing "default_datastore" for this tenant
+        if datastore_url == existing_default_ds_url:
+            return None
+
+    error_msg = tenant.set_default_datastore(auth_mgr.conn, datastore_url)
+    if error_msg:
+        error_info = generate_error_info(ErrorCode.INTERNAL_ERROR, error_msg)
+        return error_info
+    existing_default_ds = vmdk_utils.get_datastore_name(existing_default_ds_url) if existing_default_ds_url is not None else None
+    logging.info("Existing default_datastore %s is being changed to %s for tenant %s", existing_default_ds,
+                 default_datastore, tenant)
+
+    # create full access privilege to default_datastore
+    error_info = _tenant_access_add(name=tenant.name,
+                                    datastore=default_datastore,
+                                    allow_create=True)
+    # privilege to default_datastore already exist, no need to create
+    if error_info and error_info.code == ErrorCode.PRIVILEGE_ALREADY_EXIST:
+        logging.info(error_info.msg + " not overwriting the existing privilege")
```

Comment:
please do not use the word "priviledge" in messages. Use "access privilege" or simply "access"